src/config.rs

```diff
@@ -281,7 +287,12 @@ impl Cfg {
     pub fn find_override(&self, path: &Path) -> Result<Option<(Toolchain<'_>, OverrideReason)>> {
         let mut override_ = None;
 
-        // First check RUSTUP_TOOLCHAIN
+        // First check toolchain override from command
+        if let Some(ref name) = self.toolchain_override {
+            override_ = Some((name.to_string(), OverrideReason::Environment));
```

Comment:
I'd like it if you added `OverrideReason::CommandLine` whose description was something like `overridden by +toolchain on the command line` since otherwise this incorrectly states that `rustup +beta show` picks the beta toolchain due to `RUST_TOOLCHAIN` being set in the environment.